### PR TITLE
Add README

### PR DESCRIPTION
**Pull request by jeremylenz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+After a successful run, it is safe to remove katello-agent from the client.
+
+## Prerequisites
+
+### Server
+(the following soon to be done for you by foreman-installer)
+
+* Message broker such as `mosquitto` running
+* smart_proxy_remote_execution_ssh configured as follows
+```
```
Comment (ekohl) [inline README.md]: You can use syntax highlighting
````suggestion
```yaml
````

File: README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+:mode: pull-mqtt
+```
+
+### Client
+* `yggdrasil` and `foreman_ygg_worker` RPMs installed
+* Registered to Katello (script will exit if client is registered directly to RHSM)
+
+## Run
+
+```
```
Comment (ekohl) [inline README.md]: Same here
````suggestion
```sh
````

File: README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+* Gets configuration from `subscription-manager config --list`
+* Writes correct configuration in config.toml based on subscription-manager config
+* Enables pull transport daemon service (yggdrasild) on the client and displays status
+
+After a successful run, it is safe to remove katello-agent from the client.
+
+## Prerequisites
+
+### Server
+(the following soon to be done for you by foreman-installer)
```
Comment (ekohl) [inline README.md]: Please open an issue for this (https://projects.theforeman.org/projects/puppet-foreman/issues/new) and link to it here.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# katello-pull-transport-migrate
+
+_katello-pull-transport-migrate_ is a script which can be run on a client to migrate from katello-agent to REX pull transport (formerly known as "pull provider.")
+
+## What it does
+* Gets configuration from `subscription-manager config --list`
+* Writes correct configuration in config.toml based on subscription-manager config
+* Enables pull transport daemon service (yggdrasild) on the client and displays status
```
Comment (ekohl) [inline README.md]:
```suggestion
* Writes correct configuration in `config.toml` based on subscription-manager config
* Enables pull transport daemon service (`yggdrasild`) on the client and displays status
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+```
+# set $SBINDIR if it's other than /usr/sbin
+./katello-pull-transport-migrate
+```
+
+## Contribute
+
+Please, open your PR and join our [community](https://theforeman.org/contribute.html)!
+
+Issues tracked via [Redmine](https://projects.theforeman.org/projects/foreman_remote_execution).
```
Comment (ekohl) [inline README.md:40]: For this to work, you need to configure Redmine. Today that happens by adding it to https://github.com/theforeman/prprocessor/blob/master/config/repos.yaml.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (jeremylenz) [reply README.md:40]: Thanks, opened https://github.com/theforeman/prprocessor/pull/168

File: README.md (hunk 3 above)
Comment (jeremylenz) [reply README.md]: Opened https://projects.theforeman.org/issues/34239

File: README.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+### Client
+* `yggdrasil` and `foreman_ygg_worker` RPMs installed
+* Registered to Katello (script will exit if client is registered directly to RHSM)
+
+## Run
+
+```sh
+# set $SBINDIR if it's other than /usr/sbin
```
Comment (ekohl) [inline README.md]: I looked at this, but I don't really understand the need for this. Here is the only use I found:
https://github.com/theforeman/katello-pull-transport-migrate/blob/0a22ca27018f854e225a07b3aeb86ce7a42fba5f/katello-pull-transport-migrate#L11
Any process you call knows how it was called. In shell it's stored in `$0`. So if you change that line to ``then re-run `$0`.`` I'd expect it to work and you can get rid of `$SBINDIR` and be fully portable, which means it also works in development environments.

In fact, you can even add branding by creating a symlink to it (`ln -s satellite-pull-transport-migrate katello-pull-transport-migrate`). You may need to change this too:
https://github.com/theforeman/katello-pull-transport-migrate/blob/0a22ca27018f854e225a07b3aeb86ce7a42fba5f/katello-pull-transport-migrate#L49
There you may need to change it to `$(basename $0)`.

Comment (ekohl) [reply README.md]: While looking at the script I found several other things that should be improved. https://github.com/theforeman/katello-pull-transport-migrate/pull/2 implements those but doesn't implement the branding mentioned here.

Comment (jeremylenz) [reply README.md]: > you can even add branding

Per [this comment](https://github.com/theforeman/foreman-packaging/pull/7210#discussion_r771455930) from Eric we're not implementing branding yet, but I will keep it in mind.

Comment (jeremylenz) [reply README.md]: I think the `$SBINDIR` stuff was left over from [this discussion](https://github.com/theforeman/foreman-packaging/pull/7210#discussion_r765537913). You're right, it doesn't really do much. Removed and updated

File: katello-pull-transport-migrate [hunk 7]
```diff
@@ -1,14 +1,9 @@
...
-# set SBINDIR to /sbin if it is not set
-if [ -z "$SBINDIR" ]; then
-    SBINDIR=/usr/sbin
-fi
-
 # check if system is registered
 subscription-manager identity > /dev/null
 if [ $? -ne 0 ]; then
-    echo "Please register with subscription-manager first, then re-run `$SBINDIR/katello-pull-transport-migrate`."
+    echo "Please register with subscription-manager first, then re-run `$0`."
```
Comment (ekohl) [inline katello-pull-transport-migrate]: I also made this change, and more serious bug fixes in #2, such as the infinite recursion you have here on this exact line.

Comment (jeremylenz) [reply katello-pull-transport-migrate]: Okay then. I'm reverting the changes to the script. Let's keep those in #2.

**Review by jturel — APPROVED** [review]
LGTM

**Pull request merged**